### PR TITLE
[hebaov2] Add resetGuardians method in SmartWallet

### PR DESCRIPTION
**Pull request by kongliangzhong**

### CODE REVIEW

File: packages/hebao_v2/contracts/base/libwallet/GuardianLib.sol [hunk 1]
```diff
@@ -108,6 +111,48 @@ library GuardianLib
...
+    {
+        wallet.verifyApproval(
+            domainSeperator,
+            SigRequirement.MAJORITY_OWNER_REQUIRED,
+            approval,
+            abi.encode(
+                RESET_GUARDIANS_TYPEHASH,
+                approval.wallet,
+                approval.validUntil,
+                newGuardians
```
Comment (Brechtpd) [inline packages/hebao_v2/contracts/base/libwallet/GuardianLib.sol:146]: Needs to be `keccak256(abi.encodePacked(newGuardians))`?

Comment (kongliangzhong) [reply packages/hebao_v2/contracts/base/libwallet/GuardianLib.sol]: Sorry for the delay, fixed.

**Comment by dong77** [comment]
Kongliang, please add some tests, especailly, test resetting guardians when there are pending guardians or when one guardian is already an existing active guardian.

**Review by Brechtpd — APPROVED** [review]

**Comment by kongliangzhong** [comment]
> Kongliang, please add some tests, especailly, test resetting guardians when there are pending guardians or when one guardian is already an existing active guardian.

OK, will do this in another PR.

**Pull request merged**